recipes/date/all/conanfile.py

```diff
@@ -46,74 +46,81 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
+        if self.options.header_only:
+            del self.options.shared
 
     def requirements(self):
         if not self.options.header_only and not self.options.use_system_tz_db:
             self.requires("libcurl/7.78.0")
 
     def validate(self):
         if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, "11")
+            check_min_cppstd(self, 11)
 
     def package_id(self):
-        if self.options.header_only:
+        if self.info.options.header_only:
             self.info.header_only()
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        cmake = CMake(self)
-        cmake.definitions["ENABLE_DATE_TESTING"] = False
-        cmake.definitions["USE_SYSTEM_TZ_DB"] = self.options.use_system_tz_db
-        cmake.definitions["USE_TZ_DB_IN_DOT"] = self.options.use_tz_db_in_dot
-        cmake.definitions["BUILD_TZ_LIB"] = not self.options.header_only
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["ENABLE_DATE_TESTING"] = False
+        tc.cache_variables["USE_SYSTEM_TZ_DB"] = self.options.use_system_tz_db
+        tc.cache_variables["USE_TZ_DB_IN_DOT"] = self.options.use_tz_db_in_dot
+        tc.cache_variables["BUILD_TZ_LIB"] = not self.options.header_only
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.get_safe("shared", False)
```

Comment:
```suggestion
```
Automatically managed by CMakeToolchain: https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html#extending-and-customizing-cmaketoolchain